### ISSUE TITLE
Tracking issue for Improving status in CAPI resources -  Phase 1

### ISSUE THREAD

**Issue by fabriziopandini** [kubernetes-sigs/cluster-api#11105]
This is a tracking issue for activities related to #10897

## Enablement
Activities that are required to unblock implementation

- [x] @fabriziopandini Implement conditions utils #10997
- [x] @fabriziopandini Add support for metav1 conditions in patch helper #11150
- [x] @sbueringer Improve Cluster Cache Tracker to allow tracking of connection problems #11247
- [x] @chrischdi MD foreground deletion #11174
- [x] @sbueringer Handle finalizers early in Reconciles#11286

## Documentation updates
- [x] @fabriziopandini add superseded by notice to old condition proposal #11189)
- [ ] @fabriziopandini update provider contracts with notices about future changes & clarify requirements about conditions in provider contracts. Note, no backport is planned
   - [x] @fabriziopandini #11190
   - [x] @fabriziopandini #11191 
   - [x] @fabriziopandini #11194
   - [x] @fabriziopandini #11195
   - [x] @fabriziopandini #11208
   - [x] @fabriziopandini #11213
   - [x] @fabriziopandini #11223 
   - [x] @fabriziopandini #11235
   - [x] @fabriziopandini #11285
   - [ ] machine pool contract

## Implementation Phase1
Focus on phase 1 of the implementation, where new field are added but not yet used by controllers

- [x] API modeling
   - [x] @fabriziopandini #11234
   - [x] @chrischdi #11317
- [x] Machine controller
  - [x] @fabriziopandini #11276
  - [x] @fabriziopandini KCP #11389
  - [x] @fabriziopandini #11393
  - [x] @sbueringer #11291
- [x] MachineHealthCheck controller
   - [x] @fabriziopandini #11290
- [x] MachineSet controller
   - [x] @chrischdi #11278
   - [x] @sbueringer #11382
   - [x] @sbueringer #11382
   - [x] @sbueringer #11381
   - [x] @sbueringer #11390
- [x] MachineDeployment controller 
   - [x] @fabriziopandini #11338
- [x] KubeadmControlPlane controller
   - [x] @fabriziopandini #11302
   - [x] @fabriziopandini #11383
   - [x] @sbueringer #11381
   - [x] @chrischdi #11335
   - [x] @fabriziopandini #11297
   - [x] @fabriziopandini #11302
   - [x] @sbueringer #11339
- [x] Cluster controller
   - [x] @fabriziopandini #11364
   - [x] @sbueringer #11247
- [x] Topology controller
   - [x] @sbueringer #11394
- [x] ClusterClass controller
  - [x] @sbueringer #11428
- [x] KubeadmConfig controller
  - [x] @fabriziopandini #11452
- [x] ClusterResourceSet, ClusterResourceSetBinding 
  - [x] @chrischdi  https://github.com/kubernetes-sigs/cluster-api/pull/11467
- [x] clusterctl describe cluster
   - [x] @fabriziopandini #11277

### Follow up / before release
Activities that could be deferred to after we get a first version of this proposal implemented 

- [x] P0: Check initializations of the status structs to catch cases like [this](https://github.com/kubernetes-sigs/cluster-api/commit/30e7aaaf576f22fe9a62572125200d9d420c1be4)
- [x] P0: Double check deprecations notices: fields that are marked as deprecated in the proposal, utils, etc https://github.com/kubernetes-sigs/cluster-api/pull/11516 (note: we used deprecation to give an early notice to users only in case of fields that are going to be removed, not for field that will continue to exist with a different semantic)
- [x] P0: Prune fields not yet implemented before release

Phase 2 activities are tracked in #11474

**Comment by sbueringer** [comment]
Feel free to assign me to "Improve Cluster Cache Tracker to allow tracking of connection problems"

**Comment by sbueringer** [comment]
Great work everyone.

Closing this issue as everything is done for phase 1!

The only exception is the MachinePool contract. I'll leave this sub-task open but it's also tracked in https://github.com/kubernetes-sigs/cluster-api/issues/9005

/close
